### ISSUE TITLE
Zoom and jumping to cells does not work quite right 

### ISSUE THREAD

**Issue by AntonBogun** [aardappel/treesheets#232]
As I am planning to expose the control of zoom to lobster I have been going through the zoom logic and I noticed that it doesn't know what to do when the cell that is jumped to is outside of the currently visible cells. 
I have been wondering what behavior I should actually implement:

1. The jumped cell simply becomes the draw root when it is outside the visible range?
    - Or should it only become the draw root as long as the draw path is not more than the current draw path?
2. The jumped-to cell is unfolded, but otherwise as little relative movement is performed as possible?
3. Some other originally intended behavior?

This is especially noticeable when trying to F3 on a search, the draw root will typically become the root cell, even though when there is only one search result and one keeps pressing F3 it can be seen that the draw root will descend towards the result until it is unfolded.

**Comment by aardappel** [comment]
I would not expose the control of zoom directly to scripting. Instead, implement actual operations (such as move the selection to so and so cell), and ensure zoom works reasonably well with it.

The other approach is to say that we shouldn't worry about the location of the selection during script execution, but always ensure at the end of script execution that the selection and the view of it make sense. This is because you don't necessarily want the view to be jumping around during execution, that could be very slow, you simply want the script to execute as fast as possible and then see the result.

> The jumped cell simply becomes the draw root when it is outside the visible range?

There is already code that does something similar when you step into children that weren't visible or minified. What is ideal is that it steps down the draw path until the new cell is visible and not minified, but not any further, showing maximum context. Making the selected cell draw root would be very confusing in many cases.

> The jumped-to cell is unfolded, but otherwise as little relative movement is performed as possible?

Again, only if strictly necessary. There should already be code in there that shows you what happens if you jump to first child, and the cell is folded.

**Comment by AntonBogun** [comment]
Right, I also wanted to add selection operations later but I did not consider I wouldn't actually need to implement separate zoom operations.
However, I do think that the ability to force zoom into the selection grid would be a nice feature to expose to scripting, though I'm not sure that one currently has a choice about this matter with the current implementation of zoom.
